mesonbuild/environment.py

```diff
@@ -510,6 +510,7 @@ def machine_info_can_run(machine_info: MachineInfo):
     if machine_info.system != detect_system():
         return False
     true_build_cpu_family = detect_cpu_family({})
+    assert machine_info.cpu_family is not None, 'called on incomplete machine_info'
```

Comment:
This is intended to catch the case where a refactoring attempts to use this e.g. in the sanity_check function, right?

Reply:
yes, this assert would have triggered for many of the changes reverted in this PR.